### ISSUE TITLE
ucp_hello_world.c/RoCE fails on ConnectX-4 Lx

### ISSUE THREAD

**Issue by vkocheganov** [openucx/ucx#978]
Simple `<UCX>/test/examples/ucp_hello_world.c` does not work on machine with ConnectX4 Lx

This is complete output from client side:

```
[victork@r-aa-fatty25 examples]$ export UCX_IB_ETH_PAUSE_ON=y && export LD_LIBRARY_PATH=$PWD/../../install/lib

[victork@r-aa-fatty25 examples]$ ibstat
CA 'mlx5_0'
        CA type: MT4117
        Number of ports: 1
        Firmware version: 14.16.1006
        Hardware version: 0
        Node GUID: 0x7cfe900300753e60
        System image GUID: 0x7cfe900300753e60
        Port 1:
                State: Active
                Physical state: LinkUp
                Rate: 40
                Base lid: 0
                LMC: 0
                SM lid: 0
                Capability mask: 0x3c010000
                Port GUID: 0x7efe90fffe753e60
                Link layer: Ethernet

[victork@r-aa-fatty25 examples]$ export UCX_NET_DEVICES=mlx5_0:1

[victork@r-aa-fatty25 examples]$ ./a.out -n r-aa-fatty26
INFO: UCP_HELLO_WORLD mode = 0 server = r-aa-fatty26 port = 13337
UCX_NET_DEVICES=mlx5_0:1
UCX_SHM_DEVICES=all
UCX_ACC_DEVICES=all
UCX_SELF_DEVICES=all
UCX_TLS=all
UCX_ALLOC_PRIO=md:sysv,md:posix,huge,md:*,mmap,heap
UCX_BCOPY_THRESH=0
UCX_RNDV_THRESH=auto
UCX_RNDV_THRESH_FALLBACK=200000
UCX_ZCOPY_THRESH=auto
UCX_BCOPY_BW=6081740800
UCX_LOG_DATA=0
UCX_MAX_WORKER_NAME=32
[0x7c893740] local address length: 224
libibverbs: ibv_create_ah failed to get output IF
[1473757928.685043] [r-aa-fatty25:2653 :0]       ib_iface.c:239  UCX  ERROR ibv_create_ah(dlid=0 sl=0 port=1 path_bits=0dgid=fe80::7efe:90ff:fe75:3e74 sgid_index=0) on mlx5_0:1 failed: File name too long
[r-aa-fatty25:2653 :0] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
==== backtrace ====
===================
Segmentation fault (core dumped)
```
this is ofed info output:
```
[victork@r-aa-fatty25 multi_threaded_echo_c++]$ ofed_info -s
MLNX_OFED_LINUX-3.3-1.0.4.0:
```

**Comment by yosefe** [comment]
@vkocheganov can you please attach full source code?
should be solved by #969

**Comment by vkocheganov** [comment]
Oh my God, how fast are you guys :-)
Attached is source code.

**Comment by vkocheganov** [comment]
[ucp_hello_world.zip](https://github.com/openucx/ucx/files/470326/ucp_hello_world.zip)

**Comment by yosefe** [comment]
@vkocheganov it's has been an outstanding issue for a while
in this case the endpoint is destroyed while it's still sending RNDV ATS message. The fix is to flush before destroying.

**Comment by vkocheganov** [comment]
The source code is attached now 
[ucp_hello_world_origin.zip](https://github.com/openucx/ucx/files/470347/ucp_hello_world_origin.zip)

**Comment by yosefe** [comment]
should be fixed by #980 + `env UCX_IB_GID_INDEX=3`

**Comment by yosefe** [comment]
```
[yosefe@r-aa-fatty26 examples]$ UCX_IB_GID_INDEX=3 UCX_NET_DEVICES=mlx5_0:1 ./a.out -n r-aa-fatty25
INFO: UCP_HELLO_WORLD mode = 0 server = r-aa-fatty25 port = 13337
features=1 &params=0x7fff60757fb0 &params.features=0x7fff60757fb8
[1473788865.351759] [r-aa-fatty26:23414:0]    ucp_context.c:654  UCX  WARN  features 0x1 &params=0x7fff60757fb0 &params->features=0x7fff60757fb8
UCX_NET_DEVICES=mlx5_0:1
UCX_SHM_DEVICES=all
UCX_ACC_DEVICES=all
UCX_SELF_DEVICES=all
UCX_TLS=all
UCX_ALLOC_PRIO=md:sysv,md:posix,huge,md:*,mmap,heap
UCX_BCOPY_THRESH=0
UCX_RNDV_THRESH=auto
UCX_RNDV_THRESH_FALLBACK=200000
UCX_ZCOPY_THRESH=auto
UCX_BCOPY_BW=6081740800
UCX_ATOMIC_MODE=device
UCX_LOG_DATA=0
UCX_MAX_WORKER_NAME=32
[0xd23c6740] local address length: 220
UCX address message was scheduled for send
[0xd23c6740] send handler called with status 0
UCX data message was received


----- UCP TEST SUCCESS ----

hhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh

---------------------------

[1473788865.404313] [r-aa-fatty26:23414:0]          mpool.c:37   UCX  WARN  object 0x1975500 was not returned to mpool ucp_requests
```

**Comment by yosefe** [comment]
fixed by #980
